pkg/virt-handler/vm_test.go

```diff
@@ -0,0 +1,107 @@
+package virthandler
+
+import (
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"k8s.io/client-go/pkg/api"
+	kubev1 "k8s.io/client-go/pkg/api/v1"
+	metav1 "k8s.io/client-go/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+)
+
+var _ = Describe("PVC", func() {
+	RegisterFailHandler(Fail)
+	logging.DefaultLogger().SetIOWriter(GinkgoWriter)
+
+	expectedPVC := kubev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolumeClaim",
+			APIVersion: "v1",
+		},
+		Spec: kubev1.PersistentVolumeClaimSpec{
+			VolumeName: "disk-01",
+		},
+		Status: kubev1.PersistentVolumeClaimStatus{
+			Phase: kubev1.ClaimBound,
+		},
+	}
+
+	source := kubev1.ISCSIVolumeSource{
+		IQN:          "iqn.2009-02.com.test:for.all",
+		Lun:          1,
+		TargetPortal: "127.0.0.1:6543",
+	}
+
+	expectedPV := kubev1.PersistentVolume{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolume",
+			APIVersion: "v1",
+		},
+		Spec: kubev1.PersistentVolumeSpec{
+			PersistentVolumeSource: kubev1.PersistentVolumeSource{
+				ISCSI: &source,
+			},
+		},
+	}
+
+	server := ghttp.NewServer()
+	server.AppendHandlers(
+		ghttp.CombineHandlers(
+			ghttp.VerifyRequest("GET", "/apis/kubevirt.io/v1alpha1/namespaces/default/pvc/test-claim"),
+			ghttp.RespondWithJSONEncoded(http.StatusOK, expectedPVC),
+		),
+		ghttp.CombineHandlers(
+			ghttp.VerifyRequest("GET", "/apis/kubevirt.io/v1alpha1/namespaces/default/pv/disk-01"),
+			ghttp.RespondWithJSONEncoded(http.StatusOK, expectedPV),
+		),
+	)
+	defer server.Close()
+
+	vm := v1.VM{}
+
+	disk := v1.Disk{
+		Type: "PersistentVolumeClaim",
+		Source: v1.DiskSource{
+			Name: "test-claim",
+		},
+		Target: v1.DiskTarget{
+			Device: "vda",
+		},
+	}
+	disk.Type = "PersistentVolumeClaim"
+
+	domain := v1.DomainSpec{}
+	domain.Devices.Disks = []v1.Disk{disk}
+	vm.Spec.Domain = &domain
+
+	restClient := getRestClient(server.URL())
+	vmCopy, err := mapPersistentVolumes(&vm, *restClient)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(len(vmCopy.Spec.Domain.Devices.Disks)).To(Equal(1))
+	newDisk := vmCopy.Spec.Domain.Devices.Disks[0]
+	Expect(newDisk.Device).To(Equal("disk"))
+	Expect(newDisk.Source.Protocol).To(Equal("iscsi"))
+	Expect(newDisk.Source.Name).To(Equal("iqn.2009-02.com.test:for.all/1"))
+})
+
+func getRestClient(url string) *rest.RESTClient {
```

Comment:
Could  you add a wrapper like kubecli.GetRESTClient(master string, config string), also for kubecli.Get(master string, config string)?

Then you can do `coreCli, err := kubecli.Get(ghttpurl, "")` and then `coreCli.CoreV1().RESTClient()`.
Then this getRestClient section can be removed. As an alternative, you can also use the CoreV1() client directly. For our TPRs we don't do it because CoreV1() does not yet work for TPRs.